beacon-chain/db/db.go

```diff
@@ -1,95 +1,21 @@
-	HeadState(ctx context.Context) (*pb.BeaconState, error)
-	GenesisState(ctx context.Context) (*pb.BeaconState, error)
-	SaveState(ctx context.Context, state *pb.BeaconState, blockRoot [32]byte) error
-	DeleteState(ctx context.Context, blockRoot [32]byte) error
-	DeleteStates(ctx context.Context, blockRoots [][32]byte) error
-	// Slashing operations.
-	ProposerSlashing(ctx context.Context, slashingRoot [32]byte) (*ethpb.ProposerSlashing, error)
-	AttesterSlashing(ctx context.Context, slashingRoot [32]byte) (*ethpb.AttesterSlashing, error)
-	SaveProposerSlashing(ctx context.Context, slashing *ethpb.ProposerSlashing) error
-	SaveAttesterSlashing(ctx context.Context, slashing *ethpb.AttesterSlashing) error
-	HasProposerSlashing(ctx context.Context, slashingRoot [32]byte) bool
-	HasAttesterSlashing(ctx context.Context, slashingRoot [32]byte) bool
-	DeleteProposerSlashing(ctx context.Context, slashingRoot [32]byte) error
-	DeleteAttesterSlashing(ctx context.Context, slashingRoot [32]byte) error
-	// Block operations.
-	VoluntaryExit(ctx context.Context, exitRoot [32]byte) (*ethpb.VoluntaryExit, error)
-	SaveVoluntaryExit(ctx context.Context, exit *ethpb.VoluntaryExit) error
-	HasVoluntaryExit(ctx context.Context, exitRoot [32]byte) bool
-	DeleteVoluntaryExit(ctx context.Context, exitRoot [32]byte) error
-	// Checkpoint operations.
-	JustifiedCheckpoint(ctx context.Context) (*ethpb.Checkpoint, error)
-	FinalizedCheckpoint(ctx context.Context) (*ethpb.Checkpoint, error)
-	SaveJustifiedCheckpoint(ctx context.Context, checkpoint *ethpb.Checkpoint) error
-	SaveFinalizedCheckpoint(ctx context.Context, checkpoint *ethpb.Checkpoint) error
-	// Archival data handlers for storing/retrieving historical beacon node information.
-	ArchivedActiveValidatorChanges(ctx context.Context, epoch uint64) (*ethpb.ArchivedActiveSetChanges, error)
-	SaveArchivedActiveValidatorChanges(ctx context.Context, epoch uint64, changes *ethpb.ArchivedActiveSetChanges) error
-	ArchivedCommitteeInfo(ctx context.Context, epoch uint64) (*ethpb.ArchivedCommitteeInfo, error)
-	SaveArchivedCommitteeInfo(ctx context.Context, epoch uint64, info *ethpb.ArchivedCommitteeInfo) error
-	ArchivedBalances(ctx context.Context, epoch uint64) ([]uint64, error)
-	SaveArchivedBalances(ctx context.Context, epoch uint64, balances []uint64) error
-	ArchivedValidatorParticipation(ctx context.Context, epoch uint64) (*ethpb.ValidatorParticipation, error)
-	SaveArchivedValidatorParticipation(ctx context.Context, epoch uint64, part *ethpb.ValidatorParticipation) error
-	// Deposit contract related handlers.
-	DepositContractAddress(ctx context.Context) ([]byte, error)
-	SaveDepositContractAddress(ctx context.Context, addr common.Address) error
-}
+type Database = iface.Database
 
 // NewDB initializes a new DB.
 func NewDB(dirPath string) (Database, error) {
-	return kv.NewKVStore(dirPath)
+	db, err := kv.NewKVStore(dirPath)
+	if err != nil {
+		return nil, err
+	}
+
+	return kafka.Wrap(db)
```

Comment:
Shouldnt this be configured using the feature flag ?

Reply:
It is, see implementation of kafka.Wrap

Reply:
maybe it should be moved out here too